addons/mail/models/ir_config_parameter.py

```diff
@@ -1,21 +1,35 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
 
 
 class IrConfigParameter(models.Model):
     _inherit = 'ir.config_parameter'
 
+    def _check_mail_catchall_allowed_domains(self, value):
+        error_message = _('Invalid value for `mail.catchall.allowed_domains` (must be a comma seperated list of domains, e.g. example.com,example.org )')
+        if not value:
```

Comment:
Well giving a void value should probably be supported as no restriction would say, to be less strict. That way we can do "no key or no value -> no validation".